### DISCUSSION TITLE
WSGI in 4.0

### DISCUSSION THREAD

**Discussion by robb-brown** [r0x0r/pywebview#1003] [Ideas]
Hi, I'm interested in getting menu bar capability and noticed that it's been added to 4.0. I'm evaluating pywebview by writing a few utility apps for mostly my own use, so it's also a good opportunity to help test. However, I noticed that WSGI has been removed, hopefully temporarily.

I might be able to help adding WSGI to the 4.0 development, but it looks like quite a bit of the plumbing has been moving around in the 4.0 branch so I thought it would be good to at least get some idea of your plans so I can write something that fits.

Are you planning to support WSGI in 4.0? Similarly to how it's done in 3?

**Comment by r0x0r** [comment]
Homegrown HTTP server has been removed in favour of bottle.py in 4.0. Would be able to use it?

**Reply by robb-brown** [reply]
Bottle works really well. I've got basic functionality working but it's still failing a few of the tests. I'll have a look at those today and then send you a pull request.

I modified the server startup so each window starts it's own again, plus there's a global one windows that don't serve apps or files can use for their JS api. Did you want to stick with one server instance? Bottle can mount multiple apps on one server, but the user couldn't mount anything on / in that case, and opening new windows would require restarting the server.
